File: logipack-hub/hub-web/src/lib/server/hubApi/dto/offices.ts
export type OfficeDto = {
	id: string;
	name: string;
	city: string;
	address: string;

	created_at?: string | null;
	updated_at?: string | null;
	deleted_at?: string | null;
};

export type OfficeListItemDto = {
	id: string;
	name: string;
	city: string;
	address: string;

	updated_at?: string | null;
};

export type ListOfficesResponseDto = {
	offices: OfficeListItemDto[];
};

export type GetOfficeResponseDto = {
	office: OfficeDto;
};

export type CreateOfficeRequestDto = {
	name: string;
	city: string;
	address: string;
};

export type CreateOfficeResponseDto = {
	office: OfficeDto;
};

export type UpdateOfficeRequestDto = {
	name?: string | null;
	city?: string | null;
	address?: string | null;
};

export type UpdateOfficeResponseDto = {
	office: OfficeDto;
};

File: logipack-hub/hub-api/src/dto/offices.rs
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Serialize, Deserialize)]
pub struct OfficeDto {
    pub id: String,
    pub name: String,
    pub city: String,
    pub address: String,

    pub updated_at: String,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct CreateOfficeRequest {
    pub name: String,
    pub city: String,
    pub address: String,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct CreateOfficeResponse {
    pub office: OfficeDto,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct ListOfficesResponse {
    pub offices: Vec<OfficeDto>,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct GetOfficeResponse {
    pub office: OfficeDto,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct UpdateOfficeRequest {
    pub name: Option<String>,
    pub city: Option<String>,
    pub address: Option<String>,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct UpdateOfficeResponse {
    pub office: OfficeDto,
}
